plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineStandardWtpProject.java

```diff
@@ -76,6 +77,10 @@ IFile getMostImportant() {
     }
     this.config = config;
     this.uiInfoAdapter = uiInfoAdapter;
+
+    // Workaround deadlock bug described in Eclipse bug (https://bugs.eclipse.org/511793)
+    // by ensuring activation of the org.eclipse.jst.j2ee bundle
+    IJ2EEFacetConstants.DYNAMIC_WEB.toLowerCase();
```

Comment:
It's surprisingly difficult to trigger bundle activations in a reliable way.  We can certainly put this code (or similar code) in multiple places though.